engine/runner/src/main/java/org/enso/runner/EnsoLibraryFeature.java

```diff
@@ -90,46 +89,31 @@ public void beforeAnalysis(BeforeAnalysisAccess access) {
               }
             }
           }
+          if (pkg.nativeLibraryDir().exists()) {
+            var nativeLibs =
+                NativeLibraryFinder.listAllNativeLibraries(pkg, FileSystem$.MODULE$.defaultFs());
+            for (var nativeLib : nativeLibs) {
+              assert nativeLib.exists() && nativeLib.isFile();
+              var dir = nativeLib.toPath().getParent().toFile();
+              assert dir.exists() && dir.isDirectory();
+              nativeLibPaths.add(dir.getAbsolutePath());
+              var current = System.getProperty("java.library.path");
+              RuntimeSystemProperties.register(
```

Comment:
I don't think this is necessary neither good idea.

Please remember that NI build runs in build time, while the `.so`, `.dll` files are about to be located in runtime. The paths are going to be different.

Can you tell me: What was failing and why you think this change is helpful? We don't need anything like this to load `.so` for `enso_parser`...

Reply:
For `enso_parser`, we have full control over explicit search for the library in [Parser$Worker$initializeLibraries](https://github.com/enso-org/enso/blob/4d4a2990a0f0867433dd08b4e7e1ff82f6a838c1/lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java#L82) - in that method, we try to explicitly load the `enso_parser` native library first with `System.loadLibrary`, and then with `System.load`.

For opencv, we don't have full control. Its native library `opencv_java470` is loaded in the static initializer of `nu.pattern.OpenCV` with `System.loadLibrary("opencv_java470")`. `OpenCV` class is used by classes in our `std-bits/image`. In JVM mode, all classes from `std-bits/image` (classes in package `org.enso.image`) are loaded by `HostClassLoader`, therefore, also `OpenCV` class is loaded by `HostClassLoader`, and so, `System.loadLibrary("opencv_java470")` delegates to [HostClassLoader.findLibrary](https://github.com/enso-org/enso/blob/51cc81104f755b75cffabab0ba58ed46f155e567/engine/runtime/src/main/java/org/enso/interpreter/runtime/HostClassLoader.java#L99). In native image, there is no `HostClassLoader`, and we have to deal with the `System.loadLibrary` call inside `OpenCV` differently.

The solution to include path to the native library inside `RuntimeSystemProperties` from NI build time was the only solution I could think of. It points to a path like `distribution/lib/Standard/Image/0.0.0-dev/polyglot/lib/...` where the native libraries are located. All the tests are currently passing, because we test NI only via cmdline, and we don't try to package the NI inside the AppImage. It is definitely not the most robust solution. I am opened to any alternatives.

**TL;DR;** Is there any other way how to hook into `System.loadLibrary` calls during runtime in NI other than setting `java.library.path` via `RuntimeSystemProperties` during NI build?

Reply:
> In native image, there is no HostClassLoader, 

Really? How comes there is no `HostClassLoader`? Of course it cannot load any new classes, but the classloader can/should/is still be there.

Reply:
This is an example of the [use of HostClassLoader](https://github.com/jaroslavtulach/enso/commit/cc6f14549092835309f78532c36ca8ffc60b9fa9). Still some problems there, but this way we should be able to get `HostClassLoader` _"into the game"_.